examples/timeseries/timeseries_classification_from_scratch.py

```diff
@@ -0,0 +1,237 @@
+
+"""
+
+
+def make_model(input_shape):
+    input_layer = keras.layers.Input(input_shape)
+
+    # This is the first 1D convolutional block.
+    # First a linear convolution is applied by sliding a certain number of filters
+    # on the input time series to transform it into a multivariate time series,
+    # whose number of dimensions is equal to the number of filters used here.
+    conv1 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(input_layer)
+    # Following the convolution, we apply a batchnormalization in order to
+    # help the network converge quickly while adding an implicit regularization.
+    conv1 = keras.layers.BatchNormalization()(conv1)
+    # Finally we add the relu activation function to inject some non-linearity
+    # to the convolutions' output.
+    conv1 = keras.layers.ReLU()(conv1)
+
+    conv2 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv1)
+    conv2 = keras.layers.BatchNormalization()(conv2)
+    conv2 = keras.layers.ReLU()(conv2)
+
+    conv3 = keras.layers.Conv1D(filters=64, kernel_size=3, padding="same")(conv2)
+    conv3 = keras.layers.BatchNormalization()(conv3)
+    conv3 = keras.layers.ReLU()(conv3)
+
+    # A global average pooling is used here.
+    # We average the time series over the whole time dimension.
+    # This would reduce drastically the number of parameters,
+    # while enabling the use of the class activation map method for interpretability.
+    # In fact the input time series with m dimensions is averaged
+    # resulting in a vector of m dimensions.
+    gap = keras.layers.GlobalAveragePooling1D()(conv3)
+
+    # The final softmax traditional classifier is used,
+    # with a number of neurons equal to the number of classes.
+    output_layer = keras.layers.Dense(num_classes, activation="softmax")(gap)
+
+    # We link the input and output layer by constructing the keras model.
+    return keras.models.Model(inputs=input_layer, outputs=output_layer)
+
+
+model = make_model(input_shape=x_train.shape[1:])
+keras.utils.plot_model(model, show_shapes=True)
+
+"""
+## Train the model
+
+"""
+
+epochs = 500
+batch_size = 128
+
+callbacks = [
+    keras.callbacks.ModelCheckpoint(
+        "best_model.h5", save_best_only=True, monitor="val_loss"
+    ),
+    keras.callbacks.ReduceLROnPlateau(
+        monitor="loss", factor=0.5, patience=50, min_lr=0.0001
+    ),
+    keras.callbacks.EarlyStopping(monitor="val_accuracy", patience=50, verbose=1),
+]
+model.compile(
+    optimizer="adam", loss="sparse_categorical_crossentropy", metrics=["accuracy"]
+)
+history = model.fit(
+    x_train,
+    y_train,
+    batch_size=batch_size,
+    epochs=epochs,
+    callbacks=callbacks,
+    validation_split=0.2,
+    verbose=1,
+)
+
+"""
+## Evaluate model on test data
+"""
+
+model = keras.models.load_model("best_model.h5")
+
+y_pred = model.predict(x_test).argmax(axis=1)
+
+print("Test accuracy", accuracy_score(y_pred, y_test))
```

Comment:
Indeed I was using `model.evaluate`, however when I used `sparse_categorical_crossentropy` with `models.load_model("best_model.h5")` I observed weird results.
It seems like the fix would be to compile the model with `sparse_categorical_accuracy` instead of `accuracy` ([link](https://github.com/tensorflow/tensorflow/issues/39341#issuecomment-677947225)).